src/supremm/supremmconf.py

```diff
@@ -41,18 +53,44 @@ def getoptions():
 def main():
     """ print out config data according to cmdline args """
     opts = getoptions()
-    conf = Config()
-    
+
+    setuplogger(opts['log'])
+
+    if opts['config']:
+        logging.debug("Using specified path: {}".format(opts['config']))
+    else:
+        logging.debug("Automatically detecting configuration path.")
+
     try:
-        section = conf.getsection(opts['section'])
-        if opts['item'] != None:
-            print section[opts['item']]
-        else:
-            print json.dumps(section, indent=4)
+        conf = Config(opts['config'])
+    except:
+        logging.error("Configuration could not be found.")
+        sys.exit(1)
 
+    if not opts['section']:
+       print(conf)
+       sys.exit(0)
+
+    try:
+        section = conf.getsection(opts['section'])
     except KeyError:
-        sys.stderr.write("Error section \"%s\" not defined in configuration file.\n" % (opts['section']))
+        logging.error("Section '{}' not defined in configuration file.".format(opts['section']))
         sys.exit(1)
 
+    if opts['item']:
+        try:
+            item = section[opts['item']]
+        except KeyError:
+            logging.error("Item '{}' not defined in section '{}'.".format(opts['item'], opts['section']))
+            sys.exit(1)
+
+        if isinstance(item, dict):
+            item = json.dumps(item, indent=4)
+ 
+        print("{} : {}".format(opts['item'], item))
```

Comment:
This looks like a breaking change. The old code would print out the requested item, whereas the new code prints out the item name and the requested item separated by " : ".

What is the reason for the change and what is the impact of this change?

Reply:
Sorry, I didn't realize this is for the python2 PR. That is the python3 version

Reply:
Just tried and it works the same for py2